### PR TITLE
feat: Handle text overflow in buttons

### PR DESCRIPTION
**Pull request by felixjung**
## Changes:

- Adds a `noOverflow` prop to disable button text overflow and wrap text with an ellipsis (off by default)

### CODE REVIEW

**Comment by connor-baer** [comment]
Sounds good! 👍

File: src/components/Button/components/RegularButton/__snapshots__/RegularButton.spec.js.snap [hunk 1]
```diff
@@ -16,9 +16,14 @@ exports[`RegularButton should have button styles 1`] = `
...
   -webkit-text-decoration: none;
   text-decoration: none;
+  overflow: hidden;
+  white-space: nowrap;
+  text-overflow: ellipsis;
   font-size: 15px;
   line-height: 24px;
   padding: calc(8px - 0px) calc(24px - 0px);
+  overflow: visible;
+  white-space: normal;
```
Comment (fernandofleury) [inline src/components/Button/components/RegularButton/__snapshots__/RegularButton.spec.js.snap]: The button is generating both styles by default, is this intended?

**Comment by mlent** [comment]
Just popping in here - did you talk to design? It would be great if we can avoid adding more props for stuff that ideally design could take a stance on, no?

**Comment by felixjung** [comment]
I intend to talk to them. Opened the PR to "fix" behavior. No with a different default behavior this doesn't make too much sense anymore.

**Comment by felixjung** [comment]
The suggested temporary process for this is to 

- talk to the designer in your squad,
- they will bring it up in the standup,
- tell you what to do,
- ideally update the WIP docs for Circuit.

I asked Daniel and he's going to get back to me once this is decided upon.

**Comment by connor-baer** [comment]
@felixjung Any updates on this?

**Comment by felixjung** [comment]
Oh god, I kind of want to merge it but am afraid there will be a large outcry from people with too much text on buttons. 😢

**Comment by connor-baer** [comment]
@felixjung Can we merge or close this PR? 😬 

It doesn't change the default behavior, only adds a new prop to hide button overflow with an ellipsis.

**Comment by felixjung** [comment]
OK. 😁 We can make a new one changing the default value of the prop. 😅

**Comment by connor-baer** [comment]
@felixjung Last time we discussed this, we decided that this should be the default value. Why do you wanna discuss this all over again?

**Comment by connor-baer** [comment]
I discussed this topic with the design team again. We decided that it's preferable to wrap the button content rather than cutting it off. The full button label is important for users to understand what the button does and while a multi-line button might break the layout, usability is more important than visuals. 

This is the current behavior, so I'm closing this PR.

**Pull request closed without merging**